std/concurrency.d

```diff
@@ -334,6 +334,14 @@ public:
         formattedWrite(sink, "Tid(%x)", cast(void*) mbox);
     }
 
+    /**
+     * Returns whether or not this Tid has concurrency messages waiting to be
+     * received.
+     */
+    bool hasMessages() @safe pure nothrow @nogc
+    {
+        return (mbox.m_localMsgs + mbox.m_sharedBox.length) > 0;
```

Comment:
Should the access to the private members `m_localMsgs` and `m_sharedBox.length` of the private member m_box be considered Mutex:
`synchronized (m_lock)`?

Reply:
`mboxFull` doesn't, but maybe I'm misinterpreting what the comment means by "local".

```d
        // Routines involving local data only, no lock needed.

        bool mboxFull() @safe @nogc pure nothrow
        {
            return m_maxMsgs && m_maxMsgs <= m_localMsgs + m_sharedBox.length;
        }
```